evm/src/trace/decoder.rs

```diff
@@ -79,6 +79,21 @@ impl CallTraceDecoder {
     /// The call trace decoder always knows how to decode calls to the cheatcode address, as well
     /// as DSTest-style logs.
     pub fn new() -> Self {
+        let hevm_abi = HARDHAT_CONSOLE_ABI
+            .functions()
+            .map(|func| (func.short_signature(), vec![func.clone()]))
+            .collect::<BTreeMap<[u8; 4], Vec<Function>>>();
+
+        let hardhat_console_abi = HEVM_ABI
+            .functions()
+            .map(|func| (func.short_signature(), vec![func.clone()]))
+            .collect::<BTreeMap<[u8; 4], Vec<Function>>>();
+
+        let functions = hevm_abi
+            .into_iter()
+            .chain(hardhat_console_abi)
+            .collect::<BTreeMap<[u8; 4], Vec<Function>>>();
```

Comment:
This whole portion to merge the ABIs feels verbose, so let me know if there's a more concise way we should do this

Reply:
I don't think you need to collect them and then chain them, should be possible to do

```rust
HARDHAT_CONSOLE_ABI
            .functions()
            .map(|func| (func.short_signature(), vec![func.clone()]))
            .chain(HEVM_ABI
            .functions()
            .map(|func| (func.short_signature(), vec![func.clone()]))).collect::<BTreeMap<_,_>();
```

I think

Reply:
Thanks! Confirmed that worked, and done in f668168 🙂